lib/daru/dataframe.rb

```diff
@@ -295,11 +295,11 @@ def initialize source, opts={}
           end
         when Hash
           create_vectors_index_with vectors, source
+          vectors_have_same_index = all_vectors_have_equal_indexes?(source)
```

Comment:
Also, the benchmarks for your script on my machine turn to be like so:

I'm using a 32 bit debian computer with Intel(R) Core(TM) i5-2410M CPU @ 2.30GHz and 4 GB RAM.
```
require 'benchmark'
require 'daru'

# Check scaling
base_n = 10000
0.upto(2) do |iscale|
  n = base_n * 2**iscale

  df_h = ('a'..'z').map { |v| v.to_sym }.reduce({}) do |h, v|
    h[v] = Daru::Vector.new(1.upto(n).to_a)
    h
  end

  df = Daru::DataFrame.new(df_h)

  Benchmark.bm do |bm|
    bm.report("dupe (n=#{n})") do
      df.dup
    end
  end
end

#        user     system      total        real
# dupe (n=10000)  0.470000   0.000000   0.470000 (  0.479517)
#        user     system      total        real
# dupe (n=20000)  0.950000   0.010000   0.960000 (  0.957177)
#        user     system      total        real
# dupe (n=40000)  1.920000   0.000000   1.920000 (  1.921814)
# [Finished in 13.1s]
```